build/xtask/src/print.rs

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::path::Path;
+
+use anyhow::{bail, Context, Error, Result};
+
+use crate::dist::PackageConfig;
+
+pub fn run(
+    cfg: &Path,
+    archive: bool,
+    image_name: Option<String>,
+) -> Result<()> {
+    if archive {
+        let config = PackageConfig::new(cfg, false, false)
+            .context("could not create build configuration")?;
+
+        let image_name = image_name.unwrap_or(String::from("default"));
+
+        let image_name = config
+            .toml
+            .image_names
+            .iter()
+            .find(|name| name == &&image_name)
+            .ok_or(Error::msg(format!("cannot find image {}", image_name)))?;
+
+        let final_path = config
+            .img_file(format!("build-{}.zip", config.toml.name), image_name);
```

Comment:
I am a *little* not happy about this one, as it's duplicating the logic that's already been duplicated twice:

* https://github.com/oxidecomputer/hubris/blob/0b29eb59b5910c0279bfb4c16913565711005c34/build/xtask/src/humility.rs#L29
* https://github.com/oxidecomputer/hubris/blob/60ecfe39308288478914c35ffbf96ba826e9fe7c/build/xtask/src/dist.rs#L579

Not 100% sure what the best way to solve this would be, though...